mnist/mnist_gcp.ipynb

```diff
@@ -0,0 +1,895 @@
+   "source": [
+    "### Check the job\n",
+    "\n",
+    "* We can use kubectl get the status of our job"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!kubectl get tfjobs -o yaml {train_name}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Deploy TensorBoard\n",
+    "\n",
+    "* You will create a Kubernetes Deployment to run TensorBoard\n",
+    "* TensorBoard will be accessible behind the Kubeflow IAP endpoint"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "tb_name = \"mnist-tensorboard\"\n",
+    "tb_deploy = f\"\"\"apiVersion: apps/v1\n",
+    "kind: Deployment\n",
+    "metadata:\n",
+    "  labels:\n",
+    "    app: mnist-tensorboard\n",
+    "  name: {tb_name}\n",
+    "  namespace: {namespace}\n",
+    "spec:\n",
+    "  selector:\n",
+    "    matchLabels:\n",
+    "      app: mnist-tensorboard\n",
+    "  template:\n",
+    "    metadata:\n",
+    "      labels:\n",
+    "        app: mnist-tensorboard\n",
+    "        version: v1\n",
+    "    spec:\n",
+    "      serviceAccount: default-editor\n",
+    "      containers:\n",
+    "      - command:\n",
+    "        - /usr/local/bin/tensorboard\n",
+    "        - --logdir={model_dir}\n",
+    "        - --port=80\n",
+    "        image: tensorflow/tensorflow:1.15.2-py3\n",
+    "        name: tensorboard\n",
+    "        ports:\n",
+    "        - containerPort: 80\n",
+    "\"\"\"\n",
+    "tb_service = f\"\"\"apiVersion: v1\n",
+    "kind: Service\n",
+    "metadata:\n",
+    "  labels:\n",
+    "    app: mnist-tensorboard\n",
+    "  name: {tb_name}\n",
+    "  namespace: {namespace}\n",
+    "spec:\n",
+    "  ports:\n",
+    "  - name: http-tb\n",
+    "    port: 80\n",
+    "    targetPort: 80\n",
+    "  selector:\n",
+    "    app: mnist-tensorboard\n",
+    "  type: ClusterIP\n",
+    "\"\"\"\n",
+    "\n",
+    "tb_virtual_service = f\"\"\"apiVersion: networking.istio.io/v1alpha3\n",
+    "kind: VirtualService\n",
+    "metadata:\n",
+    "  name: {tb_name}\n",
+    "  namespace: {namespace}\n",
+    "spec:\n",
+    "  gateways:\n",
+    "  - kubeflow/kubeflow-gateway\n",
+    "  hosts:\n",
+    "  - '*'\n",
+    "  http:\n",
+    "  - match:\n",
+    "    - uri:\n",
+    "        prefix: /mnist/{namespace}/tensorboard/\n",
+    "    rewrite:\n",
+    "      uri: /\n",
+    "    route:\n",
+    "    - destination:\n",
+    "        host: {tb_name}.{namespace}.svc.cluster.local\n",
+    "        port:\n",
+    "          number: 80\n",
+    "    timeout: 300s\n",
+    "\"\"\"\n",
+    "\n",
+    "tb_specs = [tb_deploy, tb_service, tb_virtual_service]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "k8s_util.apply_k8s_specs(tb_specs, k8s_util.K8S_CREATE_OR_REPLACE)"
```

Comment:
Here I'm getting this error.  I actually have a bug filed for a similar thing, as the Github issue summ pipeline ex hits a related issue (let me find the bug #).  For that codelab I have them run a `create clusterrolebinding sa-admin...` command.
```
ApiException: (403)
Reason: Forbidden
HTTP response headers: HTTPHeaderDict({'Audit-Id': 'f3f47933-7471-4516-8b11-66a1e1b1e414', 'Content-Type': 'application/json', 'X-Content-Type-Options': 'nosniff', 'Date': 'Wed, 12 Feb 2020 16:32:10 GMT', 'Content-Length': '405'})
HTTP response body: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"virtualservices.networking.istio.io is forbidden: User \"system:serviceaccount:kubeflow-amyu:default-editor\" cannot create resource \"virtualservices\" in API group \"networking.istio.io\" in the namespace \"kubeflow-amyu\"","reason":"Forbidden","details":{"group":"networking.istio.io","kind":"virtualservices"},"code":403}
```
Running `kubectl create clusterrolebinding sa-admin --clusterrole=cluster-admin   --serviceaccount=kubeflow-amyu:default-editor` fixed the error.

Reply:
Looks like kubeflow/manifests/issues/891

Reply:
I verified that kubeflow/manifests#891 is now fixed at
https://github.com/kubeflow/manifests/tree/096f312